File: ManagerWeb/Services/IBankAccountService.cs
using BudgetManager.ManagerWeb.Models.DTOs;
using System;
using System.Collections.Generic;

namespace BudgetManager.ManagerWeb.Services
{
    public interface IBankAccountService
    {
        IEnumerable<BankBalanceModel> GetBankAccountsBalanceToDate(DateTime? toDate);

        IEnumerable<BankAccountModel> GetAllBankAccounts();

        int AddBankAccount(BankAccountModel bankAccountViewModel);

        void UpdateBankAccount(BankAccountModel bankAccountViewModel);

        void DeleteBankAccount(int id);
    }
}

File: ManagerWeb/Typescript/Services/IBankAccountService.ts
import BankAccountViewModel from "../Model/BankAccountViewModel";

export interface IBankAccountService {
    getAllBankAccounts(): Promise<BankAccountViewModel[]>;
    updateBankAccount(model: BankAccountViewModel): Promise<void>;
    createBankAccount(model: BankAccountViewModel): Promise<void>;
    deleteBankAccount: (id: number) => Promise<void>;
}
